### DISCUSSION TITLE
React-aria's onPress event vs native onClick

### DISCUSSION THREAD

**Discussion by borasemiz** [adobe/react-spectrum#6543] [Q&A]
I am using react-aria's date picker and button functionalities in a react project. I am using a very recent version of react (v18). When I try to create a button component which uses the `useButton` hook, I have warning from react saying that `onPress` is not a native HTML attribute, and it is ignored.

```
Warning: Unknown event handler property `onPress`. It will be ignored.
```

How can I go around this without disabling the whole warning?

**Comment by reidbarber** [comment]
I see this is marked as resolved, but I'll add an answer in case anyone else comes across it in search:

`props` (what passed into the hook) doesn't need to be spread on the element, just `buttonProps` (what gets returned from the hook):

```js
function Button(props) {
  const ref = useRef(null);
  const { buttonProps } = useButton(props, ref);
  return <button {...buttonProps} ref={ref}>{props.children}</button>;
}
```

Related discussions: https://github.com/adobe/react-spectrum/discussions/4748, https://github.com/adobe/react-spectrum/discussions/5170
